### PR TITLE
Issue #5832: Add javadoc and xdoc Example for AbbreviationAsWordInName

### PR DESCRIPTION
**Pull request by jack870131**
#5832

### CODE REVIEW

**Comment by jack870131** [comment]
**CLI Test**:

```
Jack@Jack MINGW64 ~/Documents/tmp (master)
$ javac TestAbbreviationAsWordInName.java
TestAbbreviationAsWordInName.java:1: error: class MyClass is public, should be d                                                                                                                                  eclared in a file named MyClass.java
public class MyClass { // OK
       ^
1 error

Jack@Jack MINGW64 ~/Documents/tmp (master)
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="AbbreviationAsWordInName">
      <property name="severity" value="warning"/>
      <property name="tokens" value="VARIABLE_DEF,CLASS_DEF"/>
      <property name="ignoreStatic" value="false"/>
      <property name="allowedAbbreviationLength" value="0"/>
      <property name="allowedAbbreviations" value="XML,URL"/>
    </module>
  </module>
</module>

Jack@Jack MINGW64 ~/Documents/tmp (master)
$ cat TestAbbreviationAsWordInName.java
public class MyClass { // OK
  int firstNum; // OK
  int secondNUM; // violation, it allowed only 1 consecutive capital letter
  static int thirdNum; // OK, the static modifier would be checked
  static int fourthNUm; // violation, the static modifier would be checked,
                        // and only 1 consecutive capital letter is allowed
  String firstXML; // OK, XML abbreviations are allowed
  String firstURL; // OK, URL abbreviations are allowed
}
Jack@Jack MINGW64 ~/Documents/tmp (master)
$ RUN_LOCALE="-Duser.language=en -Duser.country=US"

Jack@Jack MINGW64 ~/Documents/tmp (master)
$ java $RUN_LOCALE -jar checkstyle-8.11-all.jar -c config.xml TestAbbreviationAs                                                                                                                                  WordInName.java
Starting audit...
[WARN] C:\Users\asus\Documents\tmp\TestAbbreviationAsWordInName.java:3: Abbrevia                                                                                                                                  tion in name 'secondNUM' must contain no more than '1' consecutive capital lette                                                                                                                                  rs. [AbbreviationAsWordInName]
[WARN] C:\Users\asus\Documents\tmp\TestAbbreviationAsWordInName.java:5: Abbrevia                                                                                                                                  tion in name 'fourthNUm' must contain no more than '1' consecutive capital lette                                                                                                                                  rs. [AbbreviationAsWordInName]
Audit done.
```

**Comment by jack870131** [comment]
@romani @rnveach Waiting for your review here. Please check the identifier. I don't know if there is a better name for it.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java [hunk 1]
```diff
@@ -97,15 +97,27 @@
  * and allow no abbreviations (enforce camel case phrase) and allow no abbreviations to use (camel
  * case phrase) and allow XML and URL abbreviations.
  * </p>
+ * <p>Configuration:</p>
  * <pre>
  * &lt;module name="AbbreviationAsWordInName"&gt;
  *   &lt;property name="tokens" value="VARIABLE_DEF,CLASS_DEF"/&gt;
  *   &lt;property name="ignoreStatic" value="false"/&gt;
- *   &lt;property name="allowedAbbreviationLength" value="1"/&gt;
+ *   &lt;property name="allowedAbbreviationLength" value="0"/&gt;
```
Comment (jack870131) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java]: Setting the value to 0 to match the description above (camel case phrase).

File: src/xdocs/config_naming.xml [hunk 2]
```diff
@@ -163,14 +163,27 @@
           no abbreviations (enforce camel case phrase) and
           allow no abbreviations to use (camel case phrase) and allow XML and URL abbreviations.
         </p>
+        <p>Configuration:</p>
         <source>
 &lt;module name="AbbreviationAsWordInName"&gt;
   &lt;property name="tokens" value="VARIABLE_DEF,CLASS_DEF"/&gt;
   &lt;property name="ignoreStatic" value="false"/&gt;
-  &lt;property name="allowedAbbreviationLength" value="1"/&gt;
+  &lt;property name="allowedAbbreviationLength" value="0"/&gt;
```
Comment (jack870131) [inline src/xdocs/config_naming.xml]: Setting the value to 0 to match the description above (camel case phrase).

Comment (romani) [reply src/xdocs/config_naming.xml]: good catch but please make one more example with length=1 or 2 or 3 to show user what is possible to archive by other values.
In one example please make "xxxStatic=true", please make example with validation over only variables not a class names. Make example for user specific abbreviation like `allowedAbbreviations="CSV"` and line that show that XML become illegal.

**Comment by jack870131** [comment]
**Second Example CLI Test Result:**

```

```

File: src/xdocs/config_naming.xml [hunk 3]
```diff
@@ -160,17 +160,58 @@
...
+        <pre>
+public class MyClass { // OK, ignore checking the class name
+  int firstNum; // OK
+  int secondNUm; // OK
+  int thirdNUM; // violation, it allowed only 2 consecutive
+                // capital letter
+  static int fourthNUM; // OK, variables with static modifier
+                        // would be ignored
+  String firstCSV; // OK, CSV abbreviations are allowed
+  String firstXML; // violation, XML abbreviations are not allowed
```
Comment (romani) [inline src/xdocs/config_naming.xml]: abbreviations ==> abbreviation (in both cases)
as each variable is validated separately. `allowedAbbreviations` is plural to show that it is collection, not a single value.

File: src/xdocs/config_naming.xml [hunk 4]
```diff
@@ -160,17 +160,58 @@
...
+  &lt;property name="allowedAbbreviationLength" value="1"/&gt;
+  &lt;property name="allowedAbbreviations" value="CSV"/&gt;
+&lt;/module&gt;
+        </pre>
+        <p>Code Example:</p>
+        <pre>
+public class MyClass { // OK, ignore checking the class name
+  int firstNum; // OK
+  int secondNUm; // OK
+  int thirdNUM; // violation, it allowed only 2 consecutive
```
Comment (romani) [inline src/xdocs/config_naming.xml]: "violation, it allowed only 2 consecutive" ==> "violation, found "NUM" but it is allowed only 2 consecutive"

File: src/xdocs/config_naming.xml [hunk 5]
```diff
@@ -160,17 +160,58 @@
...
+  &lt;property name="ignoreStatic" value="true"/&gt;
+  &lt;property name="allowedAbbreviationLength" value="1"/&gt;
+  &lt;property name="allowedAbbreviations" value="CSV"/&gt;
+&lt;/module&gt;
+        </pre>
+        <p>Code Example:</p>
+        <pre>
+public class MyClass { // OK, ignore checking the class name
+  int firstNum; // OK
+  int secondNUm; // OK
```
Comment (romani) [inline src/xdocs/config_naming.xml]: `// OK, abbreviation "N" is of allowed length 1`

Please make extra line to be `int secondMYNum; // violation, found "MY" but allowed length is 1`

**Comment by jack870131** [comment]
**Travis-CI Error**:

```
Caused by: org.apache.maven.reporting.MavenReportException: 
Exit code: 1 - Picked up _JAVA_OPTIONS: -Xmx2048m -Xms512m
/home/travis/build/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java:62: error: unterminated inline tag
 * {@code @author, @version, @param, @return, @throws, @exception, @see, @since, @serial,
   ^
/home/travis/build/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java:63: error: identifier expected
 * @serialField, @serialData, @deprecated}.
   ^
/home/travis/build/checkstyle/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AtclauseOrderCheck.java:44: error: element not closed: ul
 * <ul>
   ^
```

@romani @rnveach I think this error isn't caused by my changes.

**Comment by rnveach** [comment]
Will be fixed when https://github.com/checkstyle/checkstyle/pull/6107#issue-213485036 is merged

**Comment by romani** [comment]
@jack870131 , please rebase

**Comment by jack870131** [comment]
@romani @rnveach DONE

**Comment by romani** [comment]
@rnveach , ping.
Please let me know if you have no time to do this, I will merge this my self, fix is non functional.

**Comment by jack870131** [comment]
@romani @rnveach I've correct the error. Please have a look at it.

**Comment by jack870131** [comment]
@rnveach @romani

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java [hunk 6]
```diff
@@ -92,19 +92,60 @@
...
+ *   &lt;property name="allowedAbbreviationLength" value="1"/&gt;
+ *   &lt;property name="allowedAbbreviations" value="CSV"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyClass { // OK, ignore checking the class name
+ *   int firstNum; // OK, abbreviation "N" is of allowed length 1
+ *   int secondNUm; // OK
+ *   int secondMYNum; // violation, found "MY" but allowed length is 1
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java]: > allow abbreviations up to 2 consecutive capital letters 

Shouldn't this be `found "MYN" but it is allowed only 2 consecutive capital letters`?

**Comment by rnveach** [comment]
@romani Please review and approve.

**Comment by rnveach** [comment]
@jack870131 CI is failing.

**Comment by jack870131** [comment]
@rnveach 
I'm confusing with this error in travis-ci
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.8:run (ant-phase-verify) on project checkstyle: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] /home/travis/build/checkstyle/checkstyle/config/ant-phase-verify.xml:156: Checkstyle failed: Got 2 errors and 0 warnings.
[ERROR] around Ant part ...<ant antfile="config/ant-phase-verify.xml"/>... @ 8:47 in /home/travis/build/checkstyle/checkstyle/target/antrun/build-main.xml
```
I don't know whether it is relevant to my code.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java [hunk 7]
```diff
@@ -92,20 +92,61 @@
...
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class MyClass { // OK, ignore checking the class name
+ *   int firstNum; // OK, abbreviation "N" is of allowed length 1
+ *   int secondNUm; // OK
+ *   int secondMYNum; // violation, found "MYN" but it is allowed
+ *                    // only 2 consecutive capital letters
+ *   int thirdNUM; // violation, found "NUM" but it is allowed
+ *                 // only 2 consecutive capital letters
```
Comment (IAmPramod) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java]: We should modify `it is allowed only 2 consecutive capital letters` to `only 2 consecutive capital letters are allowed`

**Comment by jack870131** [comment]
@rnveach @romani Travis CI is still failing. I don't know how to deal with this kind of error. Is it due to the update of Checkstyle?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java (hunk 7 above)
Comment (jack870131) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbbreviationAsWordInNameCheck.java]: I've correct the error. Please have a check. Thanks for your review and suggestion.

**Comment by jack870131** [comment]
@romani @rnveach  How could I fix this? Or it should be fixed from the maintainer's side?

**Comment by jack870131** [comment]
@rnveach @romani Please have a look. Thanks.

**Comment by romani** [comment]
We need to have happy CI, please instead of MYN , use some good word like GET

**Comment by jack870131** [comment]
@romani @rnveach ping. I've added "MYN" into the whitelist file. Please have a look.

**Comment by romani** [comment]
@jack870131 , thanks a lot update.
Please proceed with other Checks in naming group, we appreciate your contribution

**Pull request merged**